### ISSUE TITLE
autoHeight + wrapText in print layout creates a redraw rows loop 

### ISSUE THREAD

**Issue by RuiMadeira** [ag-grid/ag-grid#5094]
**I'm submitting a ...**
```
[X] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```

**Providing a Reproducible Scenario**
Plunker: https://plnkr.co/edit/FGQ4lbMsoBJF9srv

**Current behavior**
When switching to print mode in a grid that has at least one column with autoHeight: true and wraptText: true, the grid goes on a continously re-drawing rows loop for a while.
It appears it's recalculating and re-drawing the row heights several times.
This gets worse for even more rows and even more columns with autoHeight.

**Expected behavior**
Once switched to print mode, the grid correctly calculates the row heights once and re-renders the rows only once.

**Please tell us about your environment:**
Linux Mint 20.3 Cinnamon
Visual Studio Code
NPM
AgGrid for Angular

* **AG Grid version:** 27.1.0

* **Browser:** Google Chrome

* **Language:** TypeScript 4.4.4

**Comment by makinggoodsoftware** [comment]
Hi,

Your ticket has been flagged as in_zendesk, it means that we recognize this as an issue worth investigating so we have moved it to our official support channel for customers (zendesk)

We will carry on there with the investigation and we will update you as soon as possible.

Thanks

**Comment by jornncc** [comment]
@makinggoodsoftware 

Hello, 

I'm trying to use ag-grid for printing as well to PDF and have hit this issue myself in both ag-grid v27.3 and upgraded to v28.0 yesterday, and seeing the same thing - once I add at least one columns with `wrapText: true, autoHeight: true` then the grid becomes REALLY slow to respond to `api.setDomLayout('print');`. 

I started with your plunkr and adjusted to show the problem. The more rows/columns you have the longer it takes to get the print preview up - some times it completely breaks chrome - and you loose everything (in plunkr that is). 

https://plnkr.co/edit/oEgOXUQmph4SpBI9?open=index.jsx

Should I open a new issue? since I don't think this is fixed...

Jorn

**Comment by StephenCooper** [comment]
Hi @jornncc,

I will re-open for now so that this ticket will get a second look when we get time to look at it. 

Thanks for providing a reproduction for us.

**Comment by RuiMadeira** [comment]
Thanks @jornncc! I'm also having problems in v28.0 but didn't have time yet to do another reproducible scenario.
Nice that it was re-opened. Hope we get a fix soon :slightly_smiling_face:

**Comment by jornncc** [comment]
> Hi @jornncc,
> 
> I will re-open for now so that this ticket will get a second look when we get time to look at it.
> 
> Thanks for providing a reproduction for us.

Awesome - thank you guys for creating an amazing table :-)

**Comment by kiril-matev** [comment]
Hello!

Thank you for reporting this - much appreciated!

Based on your description we've now reproduced this issue as a bug.

We have added this bug to our backlog and we are tracking it with the following reference and description:
**AG-7163 When using columns with autoHeight and wrapText in print layout, clicking a cell or scrolling the grid is slow and less responsive**

We try to fix bugs from one release to the next, so this should be fixed in the next release or the one after if it was raised too close to the next release date.

You can see the next release date on the product pipeline page:
https://www.ag-grid.com/ag-grid-pipeline/

**Now that this issue is recorded in our backlog, we will soon close the GitHub issue because we will not update it when this bug is fixed in a future version of AG Grid. The best way to get an update on this item is to sign up for AG Grid new release notifications. Once a new AG Grid version is released, look for the item reference above (AG-XXXX) on the changelog page to see if it's resolved in that version:
https://www.ag-grid.com/ag-grid-changelog/**

Thanks again for bringing this up with us - we appreciate your time!
